azext_iot/iothub/commands_device_identity.py

```diff
@@ -0,0 +1,68 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from typing import List, Optional
+from azext_iot.common.shared import DeviceAuthType
+from azext_iot.iothub.providers.device_identity import DeviceIdentityProvider
+from knack.log import get_logger
+
+logger = get_logger(__name__)
+
+
+def iot_edge_hierarchy_create(
+    cmd,
+    devices: Optional[List[List[str]]] = None,
+    config_file: Optional[str] = None,
+    visualize: Optional[bool] = False,
+    clean: Optional[bool] = False,
+    device_auth_type: Optional[DeviceAuthType] = None,
+    root_cert_path: Optional[str] = None,
+    root_key_path: Optional[str] = None,
+    bundle_output_path: Optional[str] = None,
+    hub_name: Optional[str] = None,
+    resource_group_name: Optional[str] = None,
+    login: Optional[str] = None,
+    auth_type_dataplane: Optional[str] = None,
+):
+    device_identity_provider = DeviceIdentityProvider(
+        cmd=cmd,
+        hub_name=hub_name,
+        rg=resource_group_name,
+        login=login,
+        auth_type_dataplane=auth_type_dataplane,
+    )
+    return device_identity_provider.create_edge_hierarchy(
+        devices=devices,
+        config_file=config_file,
+        clean=clean,
+        visualize=visualize,
+        auth_type=device_auth_type,
+        root_cert_path=root_cert_path,
+        root_key_path=root_key_path,
+        output_path=bundle_output_path,
+    )
+
+
+def iot_delete_devices(
```

Comment:
May remove this provider method until we decide on a suitable namespace for the command, like `az iot hub device-identity clean/delete --all` etc.

Reply:
Since it takes a list of device Id's (its not always all devices) I could see it being useful now, as in `az iot hub device-identity delete` today takes a single deviceId , we could flex it to take 1 or more values. Someone could delete 1 or 1000 in a single command execution. Have to think about the etag though. 2 immediate thoughts are to keep an optional etag single value like now (where it would be useful for a single device delete), or also flex etag to take 1 or more values, where the etag value index corresponds to the device id value index. If the length of values mismatch raise an error.

Reply:
Hmm..we could use `eTag` as `*` in this command for the user (letting users know it won't check eTags, of course) or maybe support eTags as another param like `az iot hub device-identity delete --device id=test eTag=_etag_`

Reply:
For the first idea, right now for the device identity delete command, if the user does not provide an etag then '*' will be assumed - are you thinking if multiple device id's are provided, then the only choice is '*' + let users know of that etag behavior change?

For the second one, an initial though is it _could_ be confusing have both `--device-id` and `--device k=v` styles but we we can simmer on it.

We don't need to make a decision in this PR.

Reply:
Have a note to follow-up on the implementation of this command to allow users to delete multiple device-identities at once.